devel/smake/Portfile

```diff
@@ -1,41 +1,56 @@
+long_description	    Smake is a highly portable make program with automake \
+				        features. It is currently mainly targeted to be used \
+				        with the makefiles system.
+
+homepage		        https://sourceforge.net/projects/s-make/
+master_sites            sourceforge:schilytools
+
+checksums               sha1    876b30c2f63e1ba7d682d0cf58789d26a527f070 \
+                        rmd160  9ad475b67122f487844162ea2e6e3659ea25579f \
+                        sha256  7c93f43d22c6e3de0eafe018ceb26ddd5cc6c543dc301f307d1e6ca313248aa4 \
+                        size    4952788
+
+distname                schily-2021-09-18
+
+use_bzip2		        yes
+
+post-extract {
+                        copy ${filespath}/Gmake.smake ${worksrcpath}
+}
 
-name			smake
-version			1.2.5
-categories		devel
-platforms		darwin
-maintainers		nomaintainer
-license			CDDL
-description		highly portable make program with automake features
-long_description	Smake is a highly portable make program with automake \
-				features. It is currently mainly targeted to be used \
-				with the makefiles system. 
+# https://trac.macports.org/ticket/31616
+compiler.blacklist      llvm-gcc-4.2 macports-llvm-gcc-4.2 {clang < 300}
 
-homepage		https://sourceforge.net/projects/s-make/
-master_sites	sourceforge:project/s-make
-use_bzip2		yes
+depends_build-append    port:gmake
 
-checksums           rmd160  f683a4051273b3e24e45a2b435ddb23509db55bf \
-                    sha256  27566aa731a400c791cd95361cc755288b44ff659fa879933d4ea35d052259d4
+build.type		        gnu
+build.cmd               ${worksrcpath}/Gmake.smake
+build.args-append       CFLAGS=${configure.cflags} [get_canonical_archflags cc]
+                        
+use_parallel_build      no
 
-use_configure	no
+# destroot.args-append    CC=${configure.cc}
 
-# https://trac.macports.org/ticket/31616
-compiler.blacklist-append llvm-gcc-4.2 {clang < 318}
-
-build.type		gnu
-build.args      CC=${configure.cc}
-use_parallel_build  no
-
-destroot.args       CC=${configure.cc}
-destroot.destdir	INS_BASE=${destroot}${prefix} \
-					MANDIR=man \
-					INSUSR=`id -u` INSGRP=`id -g`
-post-destroot	{
-	file delete -force ${destroot}${prefix}/share/man/man5 \
-		${destroot}${prefix}/lib ${destroot}${prefix}/include
+destroot {
+                        set smakePath $(glob -directory ${worksrcpath}/smake/OBJ smake)
+                        xinstall $(smakePath) ${destroot}${prefix}/bin
+                        set smakeMan $(glob -directory ${worksrcpath}/smake/OBJ smake.1)
+                        xinstall -m 755 ${smakeMan} ${destroot}${prefix}/share/man/man1
 }
 
-livecheck.url       https://sourceforge.net/projects/s-make/files/
-livecheck.regex     ${name}-(\\d+(?:\\.\\d+)*)${extract.suffix}
+livecheck.name          schilytools
+livecheck.distname      schily-2021-09-18
+# livecheck.url           https://sourceforge.net/api/file/index/project-name/${livecheck.name}/rss
```

Comment:
Shall it be removed?

Reply:
That will not fix it.  If you remove the `livecheck.regex` statement in the Portfile and run `port livecheck smake`, one sees `Error: cannot check if smake was updated (regex didn't match)`.  Someone who really knows regex could easily fix it by updating the `livecheck.regex` statement.

Reply:
I fixed the regex and the livecheck.  Here is the updated code:
```

livecheck.name          schilytools
livecheck.version       2021-09-18
livecheck.distname      schily-${livecheck.version}
# livecheck.url           https://sourceforge.net/api/file/index/project-name/${livecheck.name}/rss
livecheck.regex         schily-(\\d+(?:\\-\\d+)*)
```

Reply:
Below is the fix for livecheck.  Since the port version is NOT in the name of the distribution file.  (e.g. version 1.6 is the latest version of `smake` but the latest distribution file uses a date instead -- 2021-09-18), anyone who updates the port because of a new version of `smake` must remember to manually update `livecheck.version`.  
```
livecheck.name          schilytools
livecheck.version       2021-09-18
livecheck.distname      schily-${livecheck.version}
livecheck.regex         schily-(\\d+(?:\\-\\d+)*)
```

Reply:
Fixed and forced pushed.

Reply:
@RobK88 are you sure that livecheck works? Try to run it as `port -d livecheck smake`. I do have this output:
```
DEBUG: Executing org.macports.main (smake)
DEBUG: livecheck phase started at Thu May 18 22:09:00 CEST 2023
DEBUG: Executing org.macports.livecheck (smake)
DEBUG: Port (livecheck) version is 2023-04-19
DEBUG: Loading the defaults from '/Users/catap/src/macports-ports/_resources/port1.0/livecheck/fallback.tcl'
DEBUG: Checking time since last reclaim run
```